### PR TITLE
Tests for RWTapping aggregates

### PR DESCRIPTION
**Pull request by debs-sifive**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Internal or build-related (includes code refactoring/cleanup)

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x`, `3.6.x`, or `5.x` depending on impact, API modification or big change: `6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by debs-sifive** [comment]
Draft PR b/c some of the firrtl generated is hitting a bug in firtool (which @dtzSiFive is working on). Not sure if we want to hold off on merging these tests until the output they emit are fully supported by firtool.

**Comment by dtzSiFive** [comment]
I believe these should all work as of latest firtool ( as of https://github.com/llvm/circt/commit/8c83a96372b3a1bb206e753b65999661a766f694 ), FWIW!

**Comment by jackkoenig** [comment]
I think the tests are fine since they are unit tests, although integration tests would also be handy.

File: src/test/scala/chiselTests/BoringUtilsTapSpec.scala [hunk 1]
```diff
@@ -228,4 +228,68 @@ class BoringUtilsTapSpec extends ChiselFlatSpec with ChiselRunners with Utils wi
...
+        val outRWBundleProbe = IO(probe.RWProbe(new MiniBundle))
+        val outElem = IO(probe.RWProbe(Bool()))
+        probe.define(outRWBundleProbe, BoringUtils.rwTap(child.b))
+        probe.define(outElem, outRWBundleProbe.x)
+      }
+    )
+    matchesAndOmits(chirrtl)(
+      "wire b : { x : UInt<1>}",
+      "define bore = rwprobe(b.x)",
+      "define bore_1 = rwprobe(b)",
```
Comment (jackkoenig) [inline src/test/scala/chiselTests/BoringUtilsTapSpec.scala:257]: I don't love the reliance on a default name coming from BoringUtils, is the use of BoringUtils really necessary or can we test the same behavior by just having the probe IOs on Child?

Comment (jackkoenig) [reply src/test/scala/chiselTests/BoringUtilsTapSpec.scala:257]: Or put another way, are we trying to test the behavior of BoringUtils here or just basic tapping of aggregate children? It appears to me it's more about the latter and the use of BoringUtils is just making the test a little more concise?

Comment (debs-sifive) [reply src/test/scala/chiselTests/BoringUtilsTapSpec.scala:257]: > are we trying to test the behavior of BoringUtils here or just basic tapping of aggregate children?

Kind of both... the new tap API is implemented in BoringUtils and relies on bores, which is why bores are being created.

Comment (jackkoenig) [reply src/test/scala/chiselTests/BoringUtilsTapSpec.scala:257]: I think I missed that this test is in the file `BoringUtilsTapSpec`, if that's what we're testing, that's what we should be testing 😅

**Review by jackkoenig — APPROVED** [review]

**Pull request merged**